File: Emscripten/factorial_example/factorial.cpp
#include <stdio.h>

int factorial(int n) {
  if (n == 0) {
    return 1;
  } else {
    return n * factorial(n - 1);
  }
}

int main() {
  int num;
  printf("Enter a number: ");
  scanf("%d", &num);

  int result = factorial(num);
  printf("The factorial of %d is %d\n", num, result);

  return 0;
}

File: Emscripten/factorial_example/factorial.js
// Import the WebAssembly module
const WASM = import('./factorial.wasm');

async function calculateFactorial() {
  const number = document.getElementById('numberInput').value;
  const intNumber = parseInt(number);

  // Wait for the WebAssembly module to load
  const { factorial } = await WASM;

  // Call the C++ function from JavaScript
  const result = factorial(intNumber);

  // Display the result
  document.getElementById('result').textContent = `The factorial of ${number} is ${result}`;
}

